### DISCUSSION TITLE
Cask is unreadable

### DISCUSSION THREAD

**Discussion by vr8hub** [Homebrew/discussions#113] [Everyday usage] [answered]
I did a `brew update` a few days ago and am now getting an error on almost every `brew` command, including `brew doctor`. I just updated again to see if it might have been fixed, but no go.

`brew --version` yields (on Mojave 10.14.6):
Homebrew 2.5.8
Homebrew/homebrew-core (git revision 388070; last commit 2020-11-03)
Homebrew/homebrew-cask (git revision 462399; last commit 2020-11-03)

The error I'm getting is:
Error: Cask 'java' is unreadable: undefined method `undent' for #<String:0x00007fc391172d78>

Again, I get this even on `brew doctor`, so I'm uncertain of what's wrong or how to fix it. Any ideas?

**Comment by SMillerDev** [comment]
did you ever pin the java cask?

**Comment by vr8hub** [comment]
Not to my knowledge. But I can't get any cask commands to work, either, they give the same error. Any command I would run to try to figure out what's wrong isn't working because of the error.

EDIT: I should have said "remembrance" instead of "knowledge", i.e. I don't remember ever doing so.

**Reply by SMillerDev** [reply]
@Homebrew/cask ideas?

**Comment by vr8hub** [comment]
A `brew list --pinned` returns nothing. I tried adding `--cask` but it says it's mutually exclusive. There's not a `--pinned` option on the `brew cask list` command. How would I tell if the cask was pinned?

**Comment by vitorgalvao** [comment, marked as answer]
> How would I tell if the cask was pinned?

Casks can’t be pinned. It doesn’t work with the software model of GUI apps which can update themselves (which we don’t want to break).

You’re experiencing [a long-standing bug](https://github.com/Homebrew/homebrew-cask/issues/62223). To quickly work around it, run `brew uninstall homebrew/cask/java; rm -rf "$(brew --prefix)/Caskroom/java"`. You may need to `brew install openjdk` afterwards (it’s now a formula).

**Reply by SMillerDev** [reply]
The formula is `brew install openjdk` (for anyone that reads this later)

**Reply by vitorgalvao** [reply]
> The formula is `brew install openjdk` (for anyone that reads this later)

Isn’t `java` an alias for that? Either way, I agree that’s the best one to mention. Fixed the post.

**Comment by vr8hub** [comment]
Thanks! I had already tried the uninstall; that didn't help. Apparently the `rm` did the trick. I didn't even know where to look for the cask; now I do!
